src/cls/fifo/cls_fifo_types.h

```diff
@@ -309,11 +321,40 @@ struct info {
   std::int64_t min_push_part_num{0};
   std::int64_t max_push_part_num{-1};
 
-  std::string head_tag;
-  std::map<int64_t, std::string> tags;
-
-  std::multimap<int64_t, journal_entry> journal;
+  // Use a vector rather than a flat_set so journal entries are
+  // processed in the order they were added.
+  std::vector<journal_entry> journal;
```

Comment:
why not just use the hash_map with {op, part_num} as the key and entry as the value.
the reason, the find we are doing in cls_legacy_fifo and here in apply_update to collapse duplicate operations, becomes faster

Reply:
The original idea was that I wanted the journal entries to be processed in the order in which they were added.

But on reflection I think I'll change it to a `boost::container::flat_set`. Since `create` sorts before `set_head` that should give us all the guarantee we need in practice.

`unordered_set` or other has-based containers wouldn't work since they don't have a useful traversal order.

Reply:
Used a `flat_set`.